try/catch getCameraInfo() on Camera1Engine.java

EzequielAdrianM opened the pull request:
Please merge, this is a small try/catch that helps to prevent the apps using the library from crashing on devices using the old Camera1 API.

Fix Runtime Exception on devices trying to use Camera1

Fatal Exception: java.lang.RuntimeException: Fail to get camera info
       at android.hardware.Camera.getCameraInfo(Camera.java:332)
       at com.otaliastudios.cameraview.engine.Camera1Engine.collectCameraInfo(Camera1Engine.java:144)

EzequielAdrianM committed:
try/catch getCameraInfo() on Camera1Engine.java

Fix Runtime Exception on devices trying to use Camera1

Fatal Exception: java.lang.RuntimeException: Fail to get camera info
       at android.hardware.Camera.getCameraInfo(Camera.java:332)
       at com.otaliastudios.cameraview.engine.Camera1Engine.collectCameraInfo(Camera1Engine.java:144)

natario1 commented:
Wouldn't it be better to just `continue` the for loop? Maybe the next camera info succeeds.

EzequielAdrianM committed:
Fix Runtime Exception on devices trying to use Camera1

Fatal Exception: java.lang.RuntimeException: Fail to get camera info
       at android.hardware.Camera.getCameraInfo(Camera.java:332)
       at com.otaliastudios.cameraview.engine.Camera1Engine.collectCameraInfo(Camera1Engine.java:144)

EzequielAdrianM commented:
I just modified the commit so that the loop can continue looking for other Camera info's.

EzequielAdrianM committed:
Fix mutiple getParameters() unhandled RuntimeException

Every time getParameters() is called, it can fail with RuntimeException: "getParameters failed (empty parameters)"

So the fix is to try/catch and Log.e the failure. Please tell me if you need to throw REASON_UNLNOWN exception or let the app continue.

LOGCAT CRASH:

Exception java.lang.RuntimeException: getParameters failed (empty parameters)
  at android.hardware.Camera.native_getParameters
  at android.hardware.Camera.getParameters (Camera.java:3088)
  at com.otaliastudios.cameraview.engine.Camera1Engine$3.run (Camera1Engine.java:579)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraStateOrchestrator$3.run (CameraStateOrchestrator.java:100)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator$1.call (CameraOrchestrator.java:84)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator$1.call (CameraOrchestrator.java:81)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator$3.run (CameraOrchestrator.java:152)
  at com.otaliastudios.cameraview.internal.WorkerHandler.run (WorkerHandler.java:137)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator.execute (CameraOrchestrator.java:147)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator.access$100 (CameraOrchestrator.java:34)
  at com.otaliastudios.cameraview.engine.orchestrator.CameraOrchestrator$2.run (CameraOrchestrator.java:137)
  at android.os.Handler.handleCallback (Handler.java:836)
  at android.os.Handler.dispatchMessage (Handler.java:103)
  at android.os.Looper.loop (Looper.java:203)
  at android.os.HandlerThread.run (HandlerThread.java:61)

EzequielAdrianM committed:
Catch CodecException

EzequielAdrianM committed:
Update CameraListener.java

EzequielAdrianM committed:
Update CameraView.java

EzequielAdrianM committed:
Fix Swipe Left/Right bug in CameraView.java

EzequielAdrianM committed:
Fix bug x2 CameraView.java
